### PR TITLE
Hivi leaderboard

### PR DESCRIPTION
**Pull request by emadry**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java [hunk 1]
```diff
@@ -55,13 +55,13 @@
                                                          + "']";
   private static final String GLOBAL_NAVIGATION_SELECTOR = "#globalNavigation,.site-head-container";
   private static final String MIX_CONTENT_FOOTER_ROW_SELECTOR = ".mcf-row";
-  String presentLeaderboardSelector = "div[id*='top_leaderboard']";
+  private static final String presentLeaderboardSelector = "div[id*='hivi_leaderboard']";
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java]: MAJOR Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00115)

File: src/test/java/com/wikia/webdriver/testcases/desktop/adstests/TestAdsAbcdProductsPriority.java [hunk 2]
```diff
@@ -32,8 +32,8 @@ public void adsUAPFamilyAdIsNotDisplayedOnFeaturedVideoPage() {
     AdsBaseObject ads = new AdsBaseObject(AdsDataProvider.PAGE_FV.getUrl());
     verifyAbcdHIVIAdIsNotDisplayed(ads);
 
-    Assert.assertTrue(ads.slotHasSize(AdsContent.HIVI_TOP_LB, 728, 90));
-    verifyNoUAPSizesInSlot(ads, AdsContent.HIVI_TOP_LB);
+    Assert.assertTrue(ads.slotHasSize(AdsContent.TOP_LB, 728, 90));
+    verifyNoUAPSizesInSlot(ads, AdsContent.TOP_LB);
   }
 
   private void verifyNoUAPSizesInSlot(AdsBaseObject ads, String slot) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/desktop/adstests/TestAdsAbcdProductsPriority.java]: MAJOR Remove this unused private "verifyNoUAPSizesInSlot" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 17 issues
* ![MAJOR][MAJOR] 11 major
* ![MINOR][MINOR] 6 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdsBaseObject.java#L230](https://github.com/Wikia/selenium-tests/blob/2e146b5a7fbef19d184df9162e6c9fa2c0433532/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L230): Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00117)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L231](https://github.com/Wikia/selenium-tests/blob/2e146b5a7fbef19d184df9162e6c9fa2c0433532/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L231): Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00117)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L512](https://github.com/Wikia/selenium-tests/blob/2e146b5a7fbef19d184df9162e6c9fa2c0433532/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L512): Use isEmpty() to check whether the collection is empty or not. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1155)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L545](https://github.com/Wikia/selenium-tests/blob/2e146b5a7fbef19d184df9162e6c9fa2c0433532/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L545): Use isEmpty() to check whether the collection is empty or not. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1155)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L591](https://github.com/Wikia/selenium-tests/blob/2e146b5a7fbef19d184df9162e6c9fa2c0433532/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L591): Remove this unused private "waitForProvidersChain" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L724](https://github.com/Wikia/selenium-tests/blob/2e146b5a7fbef19d184df9162e6c9fa2c0433532/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L724): Remove this unused private "triggerFMR" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L752](https://github.com/Wikia/selenium-tests/blob/2e146b5a7fbef19d184df9162e6c9fa2c0433532/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L752): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L781](https://github.com/Wikia/selenium-tests/blob/2e146b5a7fbef19d184df9162e6c9fa2c0433532/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L781): Introduce a new variable instead of reusing the parameter "duration". [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1226)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L1028](https://github.com/Wikia/selenium-tests/blob/2e146b5a7fbef19d184df9162e6c9fa2c0433532/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L1028): Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00117)
1. ![MINOR][MINOR] [AdsContent.java#L38](https://github.com/Wikia/selenium-tests/blob/2e146b5a7fbef19d184df9162e6c9fa2c0433532/src/test/java/com/wikia/webdriver/common/contentpatterns/AdsContent.java#L38): Define a constant instead of duplicating this literal "#top_leaderboard" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java (hunk 1 above)
Comment (Bielik20) [reply src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java]: 👍

**Comment by WikiaTech** [comment]
SonarQube analysis reported 16 issues
* ![MAJOR][MAJOR] 10 major
* ![MINOR][MINOR] 6 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdsBaseObject.java#L230](https://github.com/Wikia/selenium-tests/blob/18facf3f4cdf78ece035019b5c0cce86c76afc06/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L230): Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00117)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L231](https://github.com/Wikia/selenium-tests/blob/18facf3f4cdf78ece035019b5c0cce86c76afc06/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L231): Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00117)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L512](https://github.com/Wikia/selenium-tests/blob/18facf3f4cdf78ece035019b5c0cce86c76afc06/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L512): Use isEmpty() to check whether the collection is empty or not. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1155)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L545](https://github.com/Wikia/selenium-tests/blob/18facf3f4cdf78ece035019b5c0cce86c76afc06/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L545): Use isEmpty() to check whether the collection is empty or not. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1155)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L591](https://github.com/Wikia/selenium-tests/blob/18facf3f4cdf78ece035019b5c0cce86c76afc06/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L591): Remove this unused private "waitForProvidersChain" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L724](https://github.com/Wikia/selenium-tests/blob/18facf3f4cdf78ece035019b5c0cce86c76afc06/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L724): Remove this unused private "triggerFMR" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L752](https://github.com/Wikia/selenium-tests/blob/18facf3f4cdf78ece035019b5c0cce86c76afc06/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L752): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L781](https://github.com/Wikia/selenium-tests/blob/18facf3f4cdf78ece035019b5c0cce86c76afc06/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L781): Introduce a new variable instead of reusing the parameter "duration". [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1226)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L1028](https://github.com/Wikia/selenium-tests/blob/18facf3f4cdf78ece035019b5c0cce86c76afc06/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L1028): Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00117)
1. ![MINOR][MINOR] [AdsContent.java#L38](https://github.com/Wikia/selenium-tests/blob/18facf3f4cdf78ece035019b5c0cce86c76afc06/src/test/java/com/wikia/webdriver/common/contentpatterns/AdsContent.java#L38): Define a constant instead of duplicating this literal "#top_leaderboard" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 16 issues
* ![MAJOR][MAJOR] 10 major
* ![MINOR][MINOR] 6 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdsBaseObject.java#L230](https://github.com/Wikia/selenium-tests/blob/3943ac98f98331f6be5640eb78c63eb49a516015/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L230): Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00117)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L231](https://github.com/Wikia/selenium-tests/blob/3943ac98f98331f6be5640eb78c63eb49a516015/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L231): Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00117)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L512](https://github.com/Wikia/selenium-tests/blob/3943ac98f98331f6be5640eb78c63eb49a516015/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L512): Use isEmpty() to check whether the collection is empty or not. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1155)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L545](https://github.com/Wikia/selenium-tests/blob/3943ac98f98331f6be5640eb78c63eb49a516015/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L545): Use isEmpty() to check whether the collection is empty or not. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1155)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L591](https://github.com/Wikia/selenium-tests/blob/3943ac98f98331f6be5640eb78c63eb49a516015/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L591): Remove this unused private "waitForProvidersChain" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L724](https://github.com/Wikia/selenium-tests/blob/3943ac98f98331f6be5640eb78c63eb49a516015/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L724): Remove this unused private "triggerFMR" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L752](https://github.com/Wikia/selenium-tests/blob/3943ac98f98331f6be5640eb78c63eb49a516015/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L752): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L781](https://github.com/Wikia/selenium-tests/blob/3943ac98f98331f6be5640eb78c63eb49a516015/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L781): Introduce a new variable instead of reusing the parameter "duration". [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1226)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L1028](https://github.com/Wikia/selenium-tests/blob/3943ac98f98331f6be5640eb78c63eb49a516015/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L1028): Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00117)
1. ![MINOR][MINOR] [AdsContent.java#L38](https://github.com/Wikia/selenium-tests/blob/3943ac98f98331f6be5640eb78c63eb49a516015/src/test/java/com/wikia/webdriver/common/contentpatterns/AdsContent.java#L38): Define a constant instead of duplicating this literal "#top_leaderboard" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**